h2o-genmodel/src/main/java/hex/genmodel/algos/gam/GamRowToRawDataConverter.java

```diff
@@ -0,0 +1,33 @@
+package hex.genmodel.algos.gam;
+
+import hex.genmodel.GenModel;
+import hex.genmodel.easy.CategoricalEncoder;
+import hex.genmodel.easy.EasyPredictModelWrapper;
+import hex.genmodel.easy.RowData;
+import hex.genmodel.easy.RowToRawDataConverter;
+import hex.genmodel.easy.exception.PredictException;
+
+import java.util.Map;
+
+import static hex.genmodel.utils.ArrayUtils.nanArray;
+
+public class GamRowToRawDataConverter extends RowToRawDataConverter {
+  GamMojoModelBase _m;
+  public GamRowToRawDataConverter(GenModel m, Map<String, Integer> modelColumnNameToIndexMap, Map<Integer, CategoricalEncoder> domainMap, EasyPredictModelWrapper.ErrorConsumer errorConsumer, EasyPredictModelWrapper.Config config) {
+    super(m, modelColumnNameToIndexMap, domainMap, errorConsumer, config);
+    _m = (GamMojoModelBase) m;
+  }
+  
+  @Override
+  public double[] convert(RowData data, double[] rawData) throws PredictException {
+    if (_m.getPredictColumnNum() == 0) 
+      _m.setPredictColumNum(data.keySet()); // set predictColColumnNum if it has not been set.
+    if (rawData.length != _m._predictColumnNum) {
+      rawData = nanArray(_m.getTotFeatureSize());
+      rawData = super.convert(data, rawData);
+      _m.addExpandGamCols(rawData, data);
+      return rawData;
+    } else 
+      return super.convert(data, rawData);
```

Comment:
in all your tests-scenarios, does this ever really happen? I don't think it does.

Reply:
It actually does in testJavaScoring when I try to write the Junit test for GamMojo.